src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/specification/UserSpecification.kt

```diff
@@ -16,25 +22,86 @@ class UserSpecification(private val filter: UserFilter) : Specification<UserPers
     criteriaBuilder: CriteriaBuilder
   ): Predicate? {
     val predicates = mutableListOf<Predicate>()
+    fun or(vararg predicates: Predicate) = criteriaBuilder.or(*predicates)
```

Comment:
I'm pretty certain we can make the "join"more fluid and extract this out so it is generalised - just takes a bit of time